src/main/kotlin/com/ajaxproject/warehouse/service/ProductServiceImpl.kt

```diff
@@ -1,73 +1,75 @@
-    @RateLimit
-    override fun findById(id: Int): ProductDataDto {
-        return productRepository.findById(id)
-            .orElseThrow { NotFoundException("Product with id $id not found") }
-            .mapToDataDto()
+    override fun getById(id: String): ProductDataDto {
+        val mongoProduct: MongoProduct = mongoProductRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Product with id $id not found")
+        return mongoProduct.mapToDataDto()
     }
 
     override fun createProduct(createDto: ProductCreateDto): ProductDataDto {
-        val product = productRepository.save(createDto.mapToEntity())
+        val product: MongoProduct = mongoProductRepository.createProduct(createDto.mapToEntity())
         return product.mapToDataDto()
     }
 
-    override fun updateProduct(updateDto: ProductUpdateDto, id: Int): ProductDataDto {
+    override fun updateProduct(updateDto: ProductUpdateDto, id: String): ProductDataDto {
         require(id == updateDto.id) { "Mapping id is not equal to request body id" }
-        val product: Product = productRepository.findById(id)
-            .orElseThrow { NotFoundException("Product with id $id not found") }
-        product.setUpdatedData(updateDto)
-        productRepository.save(product)
+        var product: MongoProduct = mongoProductRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Product with id $id not found")
+        product = product.setUpdatedData(updateDto)
+        product = mongoProductRepository.save(product)
         return product.mapToDataDto()
     }
 
-    override fun deleteById(id: Int) {
-        productRepository.deleteById(id)
+    override fun deleteById(id: String) {
+        mongoProductRepository.deleteById(ObjectId(id))
     }
 
-    fun Product.mapToLiteDto(): ProductDataLiteDto = ProductDataLiteDto(
-        id = id,
+    fun MongoProduct.mapToLiteDto(): ProductDataLiteDto = ProductDataLiteDto(
+        id = id.toString(),
         title = title,
         price = price,
         amount = amount
     )
 
-    fun Product.mapToDataDto(): ProductDataDto = ProductDataDto(
-        id = id,
+    fun MongoProduct.mapToDataDto(): ProductDataDto = ProductDataDto(
+        id = id.toString(),
         title = title,
         price = price,
         amount = amount,
         about = about
     )
 
-    fun ProductCreateDto.mapToEntity(): Product = Product(
-        id = null,
-        title = title,
-        price = price,
+    fun ProductCreateDto.mapToEntity(): MongoProduct = MongoProduct(
+        title = title as String,
+        price = price as Double,
         amount = amount,
         about = about
     )
 
-    private fun Product.setUpdatedData(updateDto: ProductUpdateDto) {
-        title = updateDto.title as String
-        price = updateDto.price as Double
-        amount = updateDto.amount as Int
-        about = updateDto.about
+    fun MongoProduct.setUpdatedData(updateDto: ProductUpdateDto): MongoProduct {
```

Comment:
What do you think about renaming?
```suggestion
    fun MongoProduct.updateWith(updateDto: ProductUpdateDto): MongoProduct {
```

Reply:
I think that updateWith() is name more for service method, that will act with repository and another logic, and not only set fields to entity

Reply:
ok